src/server/scripts/Commands/cs_event.cpp

```diff
@@ -80,28 +81,18 @@ class event_commandscript : public CommandScript
         return true;
     }
 
-    static bool HandleEventInfoCommand(ChatHandler* handler, char const* args)
+    static bool HandleEventInfoCommand(ChatHandler* handler, Variant<Hyperlink<gameevent>, uint32> const eventId)
     {
-        if (!*args)
-            return false;
-
-        // id or [name] Shift-click form |color|Hgameevent:id|h[name]|h|r
-        char* id =  handler->extractKeyFromLink((char*)args, "Hgameevent");
-        if (!id)
-            return false;
-
-        uint32 eventId = atoul(id);
-
         GameEventMgr::GameEventDataMap const& events = sGameEventMgr->GetEventMap();
 
-        if (eventId >= events.size())
+        if (*eventId >= events.size())
         {
             handler->SendSysMessage(LANG_EVENT_NOT_EXIST);
             handler->SetSentErrorMessage(true);
             return false;
         }
 
-        GameEventData const& eventData = events[eventId];
+        GameEventData const& eventData = events[*eventId];
```

Comment:
is implicit conversion not sufficient here?

Reply:
`../src/server/scripts/Commands/cs_event.cpp:95:49: error: invalid user-defined conversion from ‘const Trinity::ChatCommands::Variant<Trinity::ChatCommands::Hyperlink<Trinity::Hyperlinks::LinkTags::gameevent>, unsigned int>’ to ‘std::vector<GameEventData>::size_type’ {aka ‘long unsigned int’} [-fpermissive]
`